python/ray/util/label_scheduling_strategies.py

```diff
@@ -0,0 +1,83 @@
+from enum import Enum
+from typing import Dict, List
+from ray.util.annotations import PublicAPI
+
+
+class LabelMatchOperator(Enum):
+    IN = "IN"
+    NOT_IN = "NOT_IN"
+    EXISTS = "EXISTS"
+    DOES_NOT_EXIST = "DOES_NOT_EXIST"
+
+
+class LabelMatchExpression:
+    """An expression used to select node by node's labels
+    Attributes:
+        key: the key of label
+        operator: IN、NOT_IN、EXISTS、DOES_NOT_EXIST,
+                  if EXISTS、DOES_NOT_EXIST, values set []
+        values: a list of label value
+    """
+
+    def __init__(self, key: str, operator: LabelMatchOperator, values: List[str]):
+        self.key = key
+        self.operator = operator
+        self.values = values
+
+
+def validate_values(values):
+    for value in values:
+        if not isinstance(value, str):
+            raise ValueError("The value of lable must be string.")
+
+
+def IN(*values):
+    validate_values(values)
+    return LabelMatchExpression(None, LabelMatchOperator.IN, list(values))
+
+
+def NOT_IN(*values):
+    validate_values(values)
+    return LabelMatchExpression(None, LabelMatchOperator.NOT_IN, list(values))
+
+
+def EXISTS():
+    return LabelMatchExpression(None, LabelMatchOperator.EXISTS, None)
+
+
+def DOES_NOT_EXIST():
+    return LabelMatchExpression(None, LabelMatchOperator.DOES_NOT_EXIST, None)
+
+
+LabelMatchExpressionsT = Dict[str, LabelMatchExpression]
+
+
+@PublicAPI(stability="beta")
+class NodeLabelSchedulingStrategy:
+    """The schedling strategy of node Affinity with labels
+    Simplified usage:
+    scheduling_strategy=NodeLabelSchedulingStrategy({
+          "region": IN("us"),
+          "gpu_type": EXISTS()
+    })
+    """
+
+    def __init__(
+        self, hard: LabelMatchExpressionsT, soft: LabelMatchExpressionsT = None
+    ):
+        self.hard = transfer_map_to_expresions(hard)
+        self.soft = transfer_map_to_expresions(soft)
+
+
+def transfer_map_to_expresions(map_expressions: LabelMatchExpressionsT):
```

Comment:
I think using an array structure would be better for implementing expressions internally, as it would allow for easier expansion in the future for scenarios where there are multiple expressions for the same key, such as {"gpu_type": [Exist(), NotIn("A100")]} without requiring major modifications.

This array structure would also result in a simpler hierarchy.